### ISSUE TITLE
process.exit() truncates piped stdout output

### ISSUE THREAD

**Issue by mkusaka** [secretlint/secretlint#1634]
**What version of secretlint are you using?**

13.0.3

**What did you do? Please include the actual steps causing the issue.**

Pipe secretlint JSON output to another process:

```bash
secretlint "examples/benchmark/targets/**/*" --format=json | wc -c
```

**What did you expect to happen?**

The full JSON output (~2.8 MB for 590 files) should be received by the piped process and parse as valid JSON.

**What actually happened? Please include the actual, raw output from secretlint.**

Only 65,536 bytes are received. The JSON is truncated and fails to parse:

```bash
$ secretlint "examples/benchmark/targets/**/*" --format=json | python3 -c "import sys,json; json.load(sys.stdin)"
json.decoder.JSONDecodeError: Unterminated string starting at: line 1 column 28210
```

The cause is `packages/secretlint/bin/secretlint.js:22` calling `process.exit(exitStatus)` immediately after `console.log(stdout)`. `process.exit()` terminates the event loop without waiting for pending I/O, so the stdout write is cut short.

| Exit method | Piped output | JSON parse |
|---|---:|---|
| `process.exit(exitStatus)` | 65,536 bytes | **Fails** |
| `process.exitCode = exitStatus` | 2,842,715 bytes | OK (590 results) |

The issue does not occur with `--output` (file write completes before exit).

Proposed fix: replace `process.exit(exitStatus)` with `process.exitCode = exitStatus`.

```diff
-    process.exit(exitStatus);
+    process.exitCode = exitStatus;
 } catch (error) {
     console.error(error);
-    process.exit(2); // fatal error
+    process.exitCode = 2; // fatal error
 }
```

**Comment by azu** [comment]
Fixed in https://github.com/secretlint/secretlint/releases/tag/v13.0.4
